pkg/tfbridge/diff.go

```diff
@@ -157,18 +157,19 @@ func makePropertyDiff(name, path string, v resource.PropertyValue, tfDiff shim.I
 	diff map[string]*pulumirpc.PropertyDiff, tfs shim.Schema, ps *SchemaInfo, finalize, rawNames bool) {
 
 	visitor := func(name, path string, v resource.PropertyValue) bool {
+		isComputedInput := ps != nil && ps.ComputedInput
```

Comment:
This is incorrect for nested properties. Let's lift it out of `visitor` and make a TODO with a comment. Here `ps` refers to top-level SchemaInfo but visitor descends into  nested properties, in fact @iwahbe was recently fixing the function that runs this visitor already for MaxItemsOne handling and I think it still might not do the right thing. We need to rewrite it sometimes. I'd have a preference reusing path based methods like LookupSchemaInfoPath to keep track of the current location in the recursion and hide away MaxItemsOne details to find the machine SchemaInfo.

Reply:
Moved up - agree on the TODO, but could you write it? Sounds like you & @iwahbe have context on the change you want to make here.